src/services/item/utils.ts

```diff
@@ -91,3 +104,93 @@ export const readPdfContent = async (source: string | URL) => {
     return '';
   }
 };
+
+export const getFieldFromMultipartForm = (
+  fields: MultipartFields,
+  key: string,
+): string | undefined => {
+  const field = fields[key];
+  if (field && !Array.isArray(field) && field.type === 'field') {
+    return field.value as string;
+  }
+};
+
+export const parseAndValidateField = <T>(
+  content: string | undefined,
+  validate: ValidateFunction<T>,
+): T | undefined => {
+  if (content) {
+    const parsedData = JSON.parse(content);
+    const isValid = validate(parsedData);
+    if (!isValid) {
+      throw new Error(validate.errors?.toString());
+    }
+    return parsedData;
+  }
+  return undefined;
+};
+
+export const getPostItemPayloadFromFormData = (
+  formData: MultipartFile | undefined,
+): {
+  item: Partial<Item> & Pick<Item, 'name' | 'type'>;
+  geolocation: Pick<ItemGeolocation, 'lat' | 'lng'> | undefined;
+  thumbnail: Readable;
+} => {
+  // if there is no formData in the request throw an error
+  if (!formData) {
+    throw new Error('Missing formData');
+  }
+
+  /**
+   * Parsing the input data from within the FormData
+   */
+  const name = getFieldFromMultipartForm(formData.fields, 'name');
+  // name should be present
+  if (!name) {
+    throw new Error('missing required name');
+  }
+  const maybeType = getFieldFromMultipartForm(formData.fields, 'type');
+  // type should be present and should be one of the available types
+  if (!maybeType || !(Object.values(ItemType) as string[]).includes(maybeType)) {
+    throw new Error('missing type or invlid type provided');
+  }
+  // here we cast as we have checked previously that it was valid
```

Comment:
After discussion, I think it's better to get the full body and validate with a single schema (probably a schema we already have).

Reply:
Yes, I created #1011